server/src/router/v1/index.ts

```diff
@@ -853,13 +854,218 @@ v1App.openapi(
 	},
 );
 
+v1App.openapi(
+	createRoute({
+		method: 'put',
+		path: '/majors/{major}',
+		tags: ['majors'],
+		summary: 'update major',
+		middleware: [authMiddleWare('admin')],
+		request: {
+			body: {
+				content: {
+					'application/json': {
+						schema: majorSchema,
+					},
+				},
+			},
+			params: z.object({
+				major: z.string(),
+			}),
+		},
+		responses: {
+			[HttpStatusCodes.NO_CONTENT]: {
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
```

Comment:
```suggestion
```

Reply:
should leave it, what if non-admin tries to access it